### DISCUSSION TITLE
[OP Stack Deployment] Error on generating the L2 config files

### DISCUSSION THREAD

**Discussion by cristianberbecaru** [ethereum-optimism/developers#351] [Deploying the OP Stack]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I followed the OP stack Docs to create a rollup on Sepolia Testnet.

After I deployed all the contracts successfully, I ran the "_forge script scripts/Deploy.s.sol:Deploy --sig 'sync()' --rpc-url $L1_RPC_URL_" and I got the following warnings/errors

![image](https://github.com/ethereum-optimism/developers/assets/32863083/164d3dda-557a-498b-8a06-093d3b4db0ce)

So, the getting-started folder from deployments folder is empty.

Meaning that, on the next step, if I run the script to generate the L2 config files ( genesis.json and rollup.json ) I get the following error:

![image](https://github.com/ethereum-optimism/developers/assets/32863083/2584d55b-90b2-471a-bcfc-2ef721f8dba4)

**Comment by Chomtana** [comment]
Which branch are you using? There are reports about failure to deploy on the latest optimism monorepo. Please try using the branch `tutorials/chain` or `op-contracts/v1.3.0` to deploy contracts (Remove old artifacts first or clone the monorepo and start over).

This log indicates that nothing has been deployed.

**Reply by cristianberbecaru** [reply]
All contracts have been deployed on Sepolia

![image](https://github.com/ethereum-optimism/developers/assets/32863083/56850bc9-13ab-4643-ad9c-b51f9a273301)

I get error when I have to generate the config files after. 

I am using tutorials/chain branch

**Reply by Chomtana** [reply]
Do you have `jq` command on your machine?

**Reply by cristianberbecaru** [reply]
Yes. I have all the dependencies. 

When I run this: **_forge script scripts/Deploy.s.sol:Deploy --sig 'sync()' --rpc-url $L1_RPC_URL_**

I get this:

![image](https://github.com/ethereum-optimism/developers/assets/32863083/a6edee86-7535-4a35-bb52-65b5cbf18072)

And my deployments/getting-started folder is missing the artifacts
![image](https://github.com/ethereum-optimism/developers/assets/32863083/fadc51e1-18ea-42ef-93f9-7f0f1b6e6379)

which will make the following command to fail: 
![image](https://github.com/ethereum-optimism/developers/assets/32863083/730d4826-09fb-4610-a663-621e71f61ffb)

**Comment by cristianberbecaru** [comment]
So .deploy file is generated with all contract addresses deployed.

![image](https://github.com/ethereum-optimism/developers/assets/32863083/6427de04-1f56-478d-ae53-8c57d803c311)


but I get this:

![image](https://github.com/ethereum-optimism/developers/assets/32863083/9adfad73-7b10-4545-89b0-b2d2abe82d08)

**Reply by cristianberbecaru** [reply]
It seems that this function can't get the transactionHash from run-latest.json file

![image](https://github.com/ethereum-optimism/developers/assets/32863083/fedde0b4-990d-464d-937f-1423a6d5cfa6)

**Reply by Chomtana** [reply]
Can you try downgradeding foundry to the version specified in https://github.com/ethereum-optimism/optimism/blob/tutorials/chain/versions.json (9e2830d7f0566e0d00b1104eeaedd5032a4e556e)

Also it would be great if you can post run-latest.json here

**Reply by cristianberbecaru** [reply]
[run-latest.json](https://github.com/ethereum-optimism/developers/files/14977578/run-latest.json)

Here is the run-latest.json

**Reply by Chomtana** [reply]
We're looking into this issue!

**Reply by Chomtana** [reply]
The newer version of OP Stack has deprecated the sync() flow

You can use branch `op-contracts/v1.3.0` to deploy contracts and generate genesis.json with `.deploy` file in the deployments folder as seen in https://github.com/upnodedev/opstack-compose/blob/4e03a46d083fcec480090a18744786418d083579/scripts/prepare.sh#L153

```
go run cmd/main.go genesis l2 \
  --deploy-config "$CONFIG_PATH"/deploy-config.json \
  --l1-deployments "/app/data/deployments/.deploy" \
  --outfile.l2 genesis.json \
  --outfile.rollup rollup.json \
  --l1-rpc "$L1_RPC_URL"
```
